services/github/github-commit-activity.service.js

```diff
@@ -103,6 +116,30 @@ export default class GitHubCommitActivity extends GithubAuthV4Service {
     })
   }
 
+  async fetchAuthorFilter({
+    interval,
+    user,
+    repo,
+    branch = 'HEAD',
+    authorFilter,
+  }) {
+    const since =
+      this.constructor.getIntervalQueryStartDate({ interval }) || undefined
+
+    return this._request({
+      url: `/repos/${user}/${repo}/commits`,
+      options: {
+        searchParams: {
+          sha: branch,
+          author: authorFilter,
+          per_page: '1',
+          since,
+        },
+      },
+      errorMessages: errorMessagesFor('repo not found'),
```

Comment:
Sorry. Another request. We don't often make changes to the interface of `BaseService`, but we've just merged https://github.com/badges/shields/pull/9233 which means this will need to become `httpErrors: httpErrorsFor('repo not found')`. Could you merge in master and update that too (sorry).